### PR TITLE
CI/TST: Use CI moto service instead of subprocess

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -25,11 +25,29 @@ jobs:
   parameters:
     name: macOS
     vmImage: macOS-10.15
+  resources:
```
Comment (lithomas1) [inline azure-pipelines.yml:28]: I think this needs to be at the top-level of the yaml file.

**Comment by mroeschke** [comment]
@lithomas1 how feasible do you think it would be to move our Windows and MacOS builds to Github Actions given the current jobs there?

For Azure Windows, appears we can't use Linux containers
https://stackoverflow.com/a/55351920
https://stackoverflow.com/a/57751095

For Azure MacOS, appears it doesn't support containers in general
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/service-containers?view=azure-devops&tabs=yaml#requirements


The goal here was to hopefully avoid flaky behavior of running a moto service in a subprocess by running it in a container instead.

**Comment by mroeschke** [comment]
Darn. For Github Actions, it looks like service containers only work for Ubuntu runners :(

https://docs.github.com/en/actions/using-containerized-services/about-service-containers#about-service-containers

**Comment by mroeschke** [comment]
@jreback thoughts about only running these s3 tests only in Linux Python 38 & 39 (and the Linux db build)? Before it was running in the Windows Python 38 & 39 builds and the Linux db build

**Comment by jreback** [comment]
> @jreback thoughts about only running these s3 tests only in Linux Python 38 & 39 (and the Linux db build)? Before it was running in the Windows Python 38 & 39 builds and the Linux db build
> 
> One can still run this locally on Windows or Mac with the subprocesses

sounds ok to me

**Comment by jreback** [comment]
ping when ready here

**Comment by mroeschke** [comment]
Ready and all green here.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@lithomas1 if any comments

**Comment by jreback** [comment]
can you merge master

**Comment by jreback** [comment]
thanks @mroeschke

**Comment by simonjayhawkins** [comment]
18 failed tests when running tests locally on ubuntu with `aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5000 ssl:default [Connect call failed ('127.0.0.1', 5000)]`

Is there something that needs to now be changed in the local dev setup also or should these tests be skipped locally?

**Comment by simonjayhawkins** [comment]
on WSL 44 failed.

**Pull request merged**